libr/util/logging.c

```diff
@@ -0,0 +1,123 @@
+/* radare - LGPL - Copyright 2007-2013 - pancake */
+
+#define LOGGING_CONFIGSTR_SIZE 512
+#define LOGGING_OUTPUTBUF_SIZE 512
+
+#include <r_core.h>
+#include <stdarg.h>
+
+// TODO: Use thread-local storage to make these variables thread-safe
+static RLoggingFuncdef cb_logfunc_print = NULL; // Function to call when outputting log string
+static int cfg_loglvl = R_LOGLVL_ERROR; // Logging level output
+static int cfg_logtraplvl = R_LOGLVL_FATALTRAP; // Logging trap level
+static bool cfg_logsrcinfo = false; // Print out debug source info with the output
+static bool cfg_logcolors = false; // Output colored log text based on level
+static char cfg_logfile[LOGGING_CONFIGSTR_SIZE] = ""; // Output text to filename
```

Comment:
I personally would move these options (except the callback of course) to RCore and do the logging to a file for example in a callback there. Basically have a default logging callback inside core, which is set for each new task by default and does some of the logic that is now implemented in `_r_logging_internal()`. What do you think?

Reply:
Hrm, I thought we wanted all this in r_util. Could you update the hackmd with the new concept, I'm not fully following.